ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -1170,5 +1183,34 @@ static Offset getIndexSegmentStartOffset(String filename) {
     }
     return new Offset(logSegmentName, Long.parseLong(startOffsetValue));
   }
+
+  /**
+   * An {@link IndexEntry} {@link Iterator} for a sealed {@link IndexSegment}. This {@link Iterator} should
+   * only be used in the compaction so that the {@link IndexSegment} should be sealed already.
+   */
+  class SealedIndexSegmentEntryIterator implements Iterator<IndexEntry> {
+    private int currentIdx = 0;
+    private ByteBuffer mmap = serEntries.duplicate();
+    private byte[] valueBuf = new byte[valueSize];
+
+    @Override
+    public boolean hasNext() {
+      return currentIdx < numberOfEntries(mmap);
```

Comment:
if this is a sealed segment, is `numberOfEntries(mmap)` a fixed value? If yes, it's worth keeping it somewhere without calculating multiple times.

Reply:
sure